kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/ParcelTest.java

```diff
@@ -65,4 +81,141 @@ void shouldRoundTrip(Set<RecordField> fields, Record record) {
+                                               @JsonProperty(required = true) ParcelContents deserializedParcelContents,
+                                               @JsonProperty(required = true) SerializationOptions serializationOptions,
+                                               Map<ParcelVersion, Exemplar> exemplars) {
+
+    }
+
+    private record NamedExemplars(ParcelSerializationExemplar exemplars, String name) {
+        Stream<NamedExemplar> flatten() {
+            ParcelContents originalRecordContents = exemplars.originalRecordContents;
+            ParcelContents deserializedParcelContents = exemplars.deserializedParcelContents;
+            return Arrays.stream(ParcelVersion.values()).map(version -> {
+                Exemplar exemplar = exemplars.exemplars.get(version);
+                return new NamedExemplar(originalRecordContents, deserializedParcelContents, version, exemplar, name, exemplars.serializationOptions);
+            });
+        }
+    }
+
+    private record NamedExemplar(ParcelContents originalRecordContents, ParcelContents deserializedParcelContents, ParcelVersion version, Exemplar exemplar, String name,
+                                 SerializationOptions options) {
+
+        public ByteBuffer serialized() {
+            return exemplar.serialized();
+        }
+
+        public String serializedBase64() {
+            return exemplar.serializedBase64();
+        }
+    }
+
+    private static final Pattern TEST_RESOURCE_FILTER = Pattern.compile("serialization/parcel/.*\\.json");
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static Stream<Arguments> exemplarStream() throws IOException {
+        return ClassPath.from(ParcelTest.class.getClassLoader()).getResources().stream()
+                .filter(ri -> TEST_RESOURCE_FILTER.matcher(ri.getResourceName()).matches())
+                .map(resourceInfo -> {
+                    try {
+                        ParcelSerializationExemplar parcelSerializationExemplar = MAPPER.reader()
+                                .readValue(resourceInfo.asByteSource().openStream(), ParcelSerializationExemplar.class);
+                        return new NamedExemplars(parcelSerializationExemplar, resourceInfo.getResourceName());
+                    }
+                    catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                })
+                .flatMap(NamedExemplars::flatten)
+                .map(exemplar -> Arguments.of(exemplar.name + " - " + exemplar.version, exemplar));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("exemplarStream")
+    void shouldDeserializeExpectedContentsFromBytes(String name, NamedExemplar exemplar) {
+        failOnVersionWithoutExemplar(exemplar);
+        Record mock = Mockito.mock(Record.class);
+        when(mock.headers()).thenReturn(new org.apache.kafka.common.header.Header[0]);
```

Comment:
the mock headers on the record are also a variable that affects the headers delivered to the consumer passed to readParcel. But I don't know if this compatibility test is a good place to test that the headers are combined.